### ISSUE TITLE
`not_using_prelude` lint lacks support for local crate

### ISSUE THREAD

**Issue by zjp-CN** [Rust-for-Linux/klint#11]
For the following [case](https://github.com/os-checker/klint-not_using_prelude/blob/2b4b5e68b73d61242ae18fdc81373ada96e729b7/tests/lint-prelude/src/main.rs), klint's not_using_prelude doesn't catch `a` is available in prelude:

```rust
fn main() {
    use a::a;
    use prelude::*;

    a();
    b();
}

pub mod a {
    pub fn a() {}
}

pub mod prelude {
    pub use super::a::a;
    pub fn b() {}
}
```

<details><summary>tidbit</summary>
<p>

To bring the check on local crate, I find [`tcx.crates()`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_middle/ty/struct.TyCtxt.html#method.crates) (used [here](https://github.com/Rust-for-Linux/klint/blob/3766c3e3060ffc7116f95ada251c8de8c55c405e/src/hir_lints/not_using_prelude.rs#L73)) iterates over dependencies only, and try appending LOCAL_CRATE in the iterator, but it turns out [`tcx.module_children`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_middle/ty/struct.TyCtxt.html#method.module_children) can't query module children on local crate at all! 
(Fair enough, rustc separates local and dependency query/analysis by design.)

```diff
        for cnum in cx
            .crates(())
            .iter()
            .copied()
            .filter(|cnum| cx.extern_crate(*cnum).is_some_and(|e| e.is_direct()))
+            .chain([LOCAL_CRATE])
```

```
$ RUSTC=klint cargo b
   Compiling dep v0.1.0 (/home/gh-zjp-CN/klint-asterinas/tests/lint-prelude/dep)
error: internal compiler error: compiler/rustc_middle/src/query/plumbing.rs:613:5: `tcx.module_children(DefId(0:0 ~ dep[828a]))` is not supported for this key;
hint: Queries can be either made to the local crate, or the external crate. This error means you tried to use it for one that's not supported.
If that's not the case, module_children was likely never assigned to a provider function.
```

</p>
</details>

**Comment by nbdd0121** [comment]
It's a intentional design choice, as it can be undesirable to require referencing via prelude inside local crate (e.g. you might want to use `super::` to refer to it).

**Comment by zjp-CN** [comment]
Ah, it makes sense. Thanks!
